third_party/nvidia/backend/compiler.py

```diff
@@ -335,7 +340,13 @@ def make_llir(self, src, metadata, options, capability):
             # Call ConcurrencySanitizerPass here, before allocating global scratch memory but after allocating tensor and shared
             passes.ttgpuir.add_concurrency_sanitizer(pm)
         passes.ttgpuir.add_allocate_global_scratch_memory(pm)
+
+        # instrumentation point here so we can override IRs above (e.g., ttir and ttgir)
+        if CUDABackend.instrumentation:
+            CUDABackend.instrumentation.patch("ttgpuir_to_llvmir", pm, mod.context)
```

Comment:
@CRobeck why it's placed before the `add_proxy_fence_insertion` pass?

Reply:
Updated. I was originally worried about the memory fences inserts interacting with the profiling but reviewing further I think you're right.